SmartDeviceLink/SDLProxy.m

```diff
@@ -166,7 +169,11 @@ - (NSString *)proxyVersion {
 
 - (SDLStreamingMediaManager *)streamingMediaManager {
     if (_streamingMediaManager == nil) {
+        if (self.displayCapabilities == nil) {
+            @throw [NSException exceptionWithName:NSInvalidArgumentException reason:@"SDLStreamingMediaManager must be accessed only after a successful RegisterAppInterfaceResponse" userInfo:nil];
+        }
         _streamingMediaManager = [[SDLStreamingMediaManager alloc] initWithProtocol:self.protocol];
```

Comment:
Can we deprecate `initWithProtocol:` and add `initWithProtocol:displayCapabilities:`? The displayCapabilities property on SMM will have to stay nullable for now, but that would clarify the actual dependencies.